term/src/iri.rs

```diff
@@ -195,6 +195,38 @@ where
         }
     }
 
+    /// Create a new suffixed IRI where `self` is used as the namespace.
+    ///
+    /// # Errors
+    ///
+    /// This method fails if `self` is itself suffixed or if the resulting IRI
+    /// is not valid.
+    ///
+    /// _Note:_ It can be enforced that an IRI is not suffixed with the
+    /// [`clone_no_suffix()`](#method.clone_no_suffix) method.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use sophia_term::iri::Iri;
+    /// let ex = Iri::<&str>::new("http://example.org/").unwrap();
+    /// let alice = ex.with_suffix("alice").unwrap();
+    ///
+    /// assert_eq!(&ex, "http://example.org/");
+    /// assert_eq!(&alice, "http://example.org/alice");
+    /// ```
+    pub fn with_suffix<V>(&self, suffix: V) -> Result<Self>
+    where
+        V: AsRef<str>,
+        TD: From<V>,
+    {
+        if self.suffix.is_some() {
+            Err(TermError::IsSuffixed)
```

Comment:
**Note on implementation:**

The `with_suffix()`-method currently throws an error if the IRI is suffixed. Theoretically, we could also use always the `Iri.ns` field and ignore `Iri.suffix`. However, I thought this would be against PolS so I decided against it.